pkg/generator/rps_test.go

```diff
@@ -0,0 +1,376 @@
+package generator
+
+import (
+	"github.com/vhive-serverless/loader/pkg/common"
+	"math"
+	"testing"
+)
+
+func TestWarmStartMatrix(t *testing.T) {
+	tests := []struct {
+		testName           string
+		experimentDuration int
+		rpsTarget          float64
+		expectedIAT        common.IATArray
+		expectedCount      []int
+	}{
+		{
+			testName:           "2min_1rps",
+			experimentDuration: 2,
+			rpsTarget:          1,
+			expectedIAT: []float64{
+				// minute 1
+				1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000,
```

Comment:
I would expect this test to lead to the same output as in trace mode with per-minute counts of `60, 60` with equidistant IAT, but in the trace tests, we expect the IATs to start with 0. One of them needs to be changed.

Reply:
Fixed in https://github.com/vhive-serverless/invitro/pull/557/commits/07875f8d12ab1b8aa750317c6c0b0a02f55a5d34.